game/src/engine/audio/AudioService.cpp

```diff
@@ -0,0 +1,60 @@
+#include "AudioService.h"
+
+#include <AL/alc.h>
+#include <AudioFile.h>
+
+#include <object_ptr.hpp>
+
+#include "engine/core/debug/Log.h"
+#include "engine/input/InputService.h"
+#include "engine/service/Service.h"
+#include "engine/service/ServiceProvider.h"
+
+// the system's default output device will be used
+const ALCchar* default_device = nullptr;
+ALCint context_attributes = 0;
+
+AudioService::AudioService()
+    : audio_device_(alcOpenDevice(default_device)),
+      audio_context_(alcCreateContext(audio_device_, 0))
+{
+}
+
+void AudioService::OnInit()
+{
+    // check audio device
+    if (!audio_device_)
+        Log::warning("[AudioService] couldn't open audio device.");
+    else
+        Log::debug("[AudioService] opened audio device successfully!");
+
```

Comment:
nit: Could you add braces to each if/else branch? I find it's easy to add extra ifs around it and cause weird things :P